# Austin Crime Reports Analysis, Pt.2

After completing the first notebook, I did some further analysis in Excel with parts of the dataset I saved to individual csv files. In particular, I separated the address column into street numbers and names to add some dimension.

In [1]:
# Importing essential libraries and configurations
%matplotlib inline

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_columns', None)

In [2]:
# Loading the dataframes
df_mur = pd.read_csv('df_mur.csv')
df_mur_cap = pd.read_csv('df_mur_cap.csv')
df_rape = pd.read_csv('df_rape.csv')

In [3]:
df_mur.report_date_time = df_mur.report_date_time.astype('datetime64')
df_mur.set_index(['report_date_time'], inplace=True)
df_mur.sort_index(inplace=True)

df_mur_cap.report_date_time = df_mur_cap.report_date_time.astype('datetime64')
df_mur_cap.set_index(['report_date_time'], inplace=True)
df_mur_cap.sort_index(inplace=True)

df_rape.report_date_time = df_rape.report_date_time.astype('datetime64')
df_rape.set_index(['report_date_time'], inplace=True)
df_rape.sort_index(inplace=True)

# Converting columns to their appropriate data types
df_mur.clearance_date = df_mur.clearance_date.astype('datetime64')
df_mur.occurred_date = df_mur.occurred_date.astype('datetime64')
df_mur.highest_offense_code = df_mur.highest_offense_code.astype('int64')

df_mur_cap.clearance_date = df_mur_cap.clearance_date.astype('datetime64')
df_mur_cap.occurred_date = df_mur_cap.occurred_date.astype('datetime64')
df_mur_cap.highest_offense_code = df_mur_cap.highest_offense_code.astype('int64')

df_rape.clearance_date = df_rape.clearance_date.astype('datetime64')
df_rape.occurred_date = df_rape.occurred_date.astype('datetime64')
df_rape.highest_offense_code = df_rape.highest_offense_code.astype('int64')

## How are violent crimes distributed...

### Question 1. Are there any addresses in Austin, known to be hotspots for violent crime? 

Note: For non-capital murder, in the following code, I am only including results that have hosted 2 incidents or more. For rape, I am only including the top 50 results.

In [4]:
print('-----------------------------------------------')
print('Addresses where violent crimes most often occur')
print('-----------------------------------------------')
print('Murder')
print('----------------------------------')
display(df_mur.address.value_counts().head(22))
print('----------------------------------')
display(df_mur.address.value_counts(normalize=True).head(22))

print('----------------------------------')
print('Capital Murder')
print('----------------------------------')
display(df_mur_cap.address.value_counts())
print('----------------------------------')
display(df_mur_cap.address.value_counts(normalize=True))
print('----------------------------------')
print('Rape')
print('----------------------------------')
display(df_rape.address.value_counts().head(50))
print('----------------------------------')
display(df_rape.address.value_counts(normalize=True).head(50))

-----------------------------------------------
Addresses where violent crimes most often occur
-----------------------------------------------
Murder
----------------------------------


8610 N LAMAR BLVD           3
4700 E RIVERSIDE DR         3
8800 N IH 35 SVRD SB        3
2501 ANKEN DR               2
2508 E RIVERSIDE DR         2
1050 SPRINGDALE RD          2
9010 GALEWOOD DR            2
10610 MORADO CIR            2
6409 SPRINGDALE RD          2
9220 N IH 35 SVRD SB        2
9601 MIDDLE FISKVILLE RD    2
6118 FAIRWAY ST             2
2400 WICKERSHAM LN          2
1137 GUNTER ST              2
6407 SPRINGDALE RD          2
833 FAIRFIELD DR            2
2713 S CONGRESS AVE         2
9200 NORTH PLZ              2
201 W 5TH ST                2
601 E 15TH ST               2
2915 S IH 35 SVRD NB        2
8912 N LAMAR BLVD           2
Name: address, dtype: int64

----------------------------------


8610 N LAMAR BLVD           0.007792
4700 E RIVERSIDE DR         0.007792
8800 N IH 35 SVRD SB        0.007792
2501 ANKEN DR               0.005195
2508 E RIVERSIDE DR         0.005195
1050 SPRINGDALE RD          0.005195
9010 GALEWOOD DR            0.005195
10610 MORADO CIR            0.005195
6409 SPRINGDALE RD          0.005195
9220 N IH 35 SVRD SB        0.005195
9601 MIDDLE FISKVILLE RD    0.005195
6118 FAIRWAY ST             0.005195
2400 WICKERSHAM LN          0.005195
1137 GUNTER ST              0.005195
6407 SPRINGDALE RD          0.005195
833 FAIRFIELD DR            0.005195
2713 S CONGRESS AVE         0.005195
9200 NORTH PLZ              0.005195
201 W 5TH ST                0.005195
601 E 15TH ST               0.005195
2915 S IH 35 SVRD NB        0.005195
8912 N LAMAR BLVD           0.005195
Name: address, dtype: float64

----------------------------------
Capital Murder
----------------------------------


815 W SLAUGHTER LN              2
7000 DECKER LN                  2
9200 NORTH PLAZA                1
12900 N IH 35 SVRD SB           1
2100 HOLLY ST                   1
612 DENSON DR                   1
5603 PORSCHE LN                 1
1200 E 52ND ST                  1
1500 BLOCK BURTON DR            1
1811 J J SEABROOK DR            1
6900 BLOCK BLESSING AVE         1
11701 METRIC BLVD               1
2812 NUECES ST                  1
3128 MANOR RD                   1
4708 WHISPERING VALLEY DR       1
360 NUECES ST                   1
2110 E 4TH ST                   1
1711 RUTLAND DR                 1
6103 HYSIDE DR                  1
1200 BLOCK E 38TH HALF ST       1
6323 SHADOW BND                 1
1105 DARVONE CIR                1
5700 CAMERON RD                 1
2904 PECAN SPRINGS RD           1
11040 KIRKOSWALD RD             1
1030 NORWOOD PARK BLVD          1
12334 COPPERFIELD DR            1
605 MASTERSON PASS              1
4600 BLOCK BANDERA RD           1
6808 MONTANA S

----------------------------------


815 W SLAUGHTER LN              0.025316
7000 DECKER LN                  0.025316
9200 NORTH PLAZA                0.012658
12900 N IH 35 SVRD SB           0.012658
2100 HOLLY ST                   0.012658
612 DENSON DR                   0.012658
5603 PORSCHE LN                 0.012658
1200 E 52ND ST                  0.012658
1500 BLOCK BURTON DR            0.012658
1811 J J SEABROOK DR            0.012658
6900 BLOCK BLESSING AVE         0.012658
11701 METRIC BLVD               0.012658
2812 NUECES ST                  0.012658
3128 MANOR RD                   0.012658
4708 WHISPERING VALLEY DR       0.012658
360 NUECES ST                   0.012658
2110 E 4TH ST                   0.012658
1711 RUTLAND DR                 0.012658
6103 HYSIDE DR                  0.012658
1200 BLOCK E 38TH HALF ST       0.012658
6323 SHADOW BND                 0.012658
1105 DARVONE CIR                0.012658
5700 CAMERON RD                 0.012658
2904 PECAN SPRINGS RD           0.012658
11040 KIRKOSWALD

----------------------------------
Rape
----------------------------------


900 BLOCK E 32ND ST                   51
700 BLOCK E 8TH ST                    29
1300 BLOCK CROSSING PL                28
12200 BLOCK N MOPAC EXPY SVRD NB      18
300 BLOCK E 6TH ST                    17
4100 BLOCK GUADALUPE ST               17
1600 BLOCK WICKERSHAM LN              16
600 BLOCK E 15TH ST                   15
500 BLOCK E 4TH ST                    15
1500 BLOCK WICKERSHAM LN              15
900 BLOCK W BEN WHITE BLVD SVRD EB    14
1500 BLOCK GROVE BLVD                 13
2100 BLOCK BURTON DR                  12
400 BLOCK E WILLIAM CANNON DR         11
5300 BLOCK N IH 35 SVRD SB            10
200 BLOCK LAVACA ST                   10
9300 BLOCK NORTHGATE BLVD             10
1100 BLOCK GUNTER ST                  10
2700 BLOCK S IH 35 SVRD NB            10
11100 BLOCK RESEARCH BLVD SVRD NB     10
9400 BLOCK N IH 35 SVRD SB            10
1600 BLOCK ROYAL CREST DR             10
4400 BLOCK E OLTORF ST                10
2400 BLOCK TOWN LAKE CIR               9
1700 BLOCK TERI 

----------------------------------


900 BLOCK E 32ND ST                   0.012397
700 BLOCK E 8TH ST                    0.007049
1300 BLOCK CROSSING PL                0.006806
12200 BLOCK N MOPAC EXPY SVRD NB      0.004375
300 BLOCK E 6TH ST                    0.004132
4100 BLOCK GUADALUPE ST               0.004132
1600 BLOCK WICKERSHAM LN              0.003889
600 BLOCK E 15TH ST                   0.003646
500 BLOCK E 4TH ST                    0.003646
1500 BLOCK WICKERSHAM LN              0.003646
900 BLOCK W BEN WHITE BLVD SVRD EB    0.003403
1500 BLOCK GROVE BLVD                 0.003160
2100 BLOCK BURTON DR                  0.002917
400 BLOCK E WILLIAM CANNON DR         0.002674
5300 BLOCK N IH 35 SVRD SB            0.002431
200 BLOCK LAVACA ST                   0.002431
9300 BLOCK NORTHGATE BLVD             0.002431
1100 BLOCK GUNTER ST                  0.002431
2700 BLOCK S IH 35 SVRD NB            0.002431
11100 BLOCK RESEARCH BLVD SVRD NB     0.002431
9400 BLOCK N IH 35 SVRD SB            0.002431
1600 BLOCK RO

### Question 2. Are there any specific streets in Austin, that are hotspots for violent crime? 

Note: For murder only, in the following code, I am only including results that >= 1%...

In [5]:
print('---------------------------------------------')
print('Streets where violent crimes most offen occur')
print('---------------------------------------------')

print('----------------------------------')
print('Murder')
print('----------------------------------')
display(df_mur.street_name.value_counts().head(10))
print('----------------------------------')
display(df_mur.street_name.value_counts(normalize=True).head(10))
print('----------------------------------')
print('Capital Murder')
print('----------------------------------')
display(df_mur_cap.street_name.value_counts())
print('----------------------------------')
display(df_mur_cap.street_name.value_counts(normalize=True))
print('----------------------------------')
print('Rape')
print('----------------------------------')
display(df_rape.street_name.value_counts())
print('----------------------------------')
display(df_rape.street_name.value_counts(normalize=True))

---------------------------------------------
Streets where violent crimes most offen occur
---------------------------------------------
----------------------------------
Murder
----------------------------------


E RIVERSIDE DR         13
N LAMAR BLVD           10
N IH 35 SVRD SB         9
SPRINGDALE RD           7
RUTLAND DR              6
S CONGRESS AVE          5
GALEWOOD DR             4
WICKERSHAM LN           4
W WILLIAM CANNON DR     4
E 12TH ST               4
Name: street_name, dtype: int64

----------------------------------


E RIVERSIDE DR         0.033766
N LAMAR BLVD           0.025974
N IH 35 SVRD SB        0.023377
SPRINGDALE RD          0.018182
RUTLAND DR             0.015584
S CONGRESS AVE         0.012987
GALEWOOD DR            0.010390
WICKERSHAM LN          0.010390
W WILLIAM CANNON DR    0.010390
E 12TH ST              0.010390
Name: street_name, dtype: float64

----------------------------------
Capital Murder
----------------------------------


N LAMAR BLVD                4
RUTLAND DR                  3
SPRINGDALE RD               2
W SLAUGHTER LN              2
N IH 35 SVRD NB             2
DECKER LN                   2
NUECES ST                   2
MONTANA ST                  1
E 4TH ST                    1
LN / BURLESON RD            1
BETHUNE AVE                 1
TEABERRY DR                 1
DARVONE CIR                 1
E OLTORF ST                 1
SHADOW BND                  1
MANOR RD                    1
KRAMER LN                   1
E WILLIAM CANNON DR         1
W 21ST ST                   1
KIRKOSWALD RD               1
HOLLY ST                    1
SUMMER DR                   1
BLOCK ELMONT DR             1
ED BLUESTEIN BLVD SB        1
NORTHGATE BLVD              1
E BEN WHITE BLVD SVRD EB    1
BLOCK BLESSING AVE          1
SCENIC DR                   1
WILLOW CREEK DR             1
N IH 35 SVRD SB             1
                           ..
REINLI ST                   1
W WILLIAM CANNON DR         1
BLOCK E 38

----------------------------------


N LAMAR BLVD                0.050633
RUTLAND DR                  0.037975
SPRINGDALE RD               0.025316
W SLAUGHTER LN              0.025316
N IH 35 SVRD NB             0.025316
DECKER LN                   0.025316
NUECES ST                   0.025316
MONTANA ST                  0.012658
E 4TH ST                    0.012658
LN / BURLESON RD            0.012658
BETHUNE AVE                 0.012658
TEABERRY DR                 0.012658
DARVONE CIR                 0.012658
E OLTORF ST                 0.012658
SHADOW BND                  0.012658
MANOR RD                    0.012658
KRAMER LN                   0.012658
E WILLIAM CANNON DR         0.012658
W 21ST ST                   0.012658
KIRKOSWALD RD               0.012658
HOLLY ST                    0.012658
SUMMER DR                   0.012658
BLOCK ELMONT DR             0.012658
ED BLUESTEIN BLVD SB        0.012658
NORTHGATE BLVD              0.012658
E BEN WHITE BLVD SVRD EB    0.012658
BLOCK BLESSING AVE          0.012658
S

----------------------------------
Rape
----------------------------------


BLOCK N IH 35 SVRD SB                 95
BLOCK N LAMAR BLVD                    80
BLOCK S CONGRESS AVE                  58
BLOCK E 32ND ST                       55
BLOCK WICKERSHAM LN                   53
BLOCK E RIVERSIDE DR                  51
BLOCK N IH 35 SVRD NB                 49
BLOCK S IH 35 SVRD NB                 42
BLOCK E 8TH ST                        40
BLOCK E 6TH ST                        37
BLOCK BURTON DR                       36
BLOCK GUADALUPE ST                    35
BLOCK E OLTORF ST                     34
BLOCK S 1ST ST                        33
BLOCK W WILLIAM CANNON DR             32
BLOCK N I 35 SB SVC RD                31
BLOCK CROSSING PL                     31
BLOCK E 4TH ST                        30
BLOCK MANCHACA RD                     29
BLOCK NORTHGATE BLVD                  27
BLOCK RUTLAND DR                      26
BLOCK N FM 620 RD                     26
BLOCK E WILLIAM CANNON DR             26
BLOCK RED RIVER ST                    25
BLOCK CAMERON RD

----------------------------------


BLOCK N IH 35 SVRD SB                 0.023092
BLOCK N LAMAR BLVD                    0.019446
BLOCK S CONGRESS AVE                  0.014098
BLOCK E 32ND ST                       0.013369
BLOCK WICKERSHAM LN                   0.012883
BLOCK E RIVERSIDE DR                  0.012397
BLOCK N IH 35 SVRD NB                 0.011911
BLOCK S IH 35 SVRD NB                 0.010209
BLOCK E 8TH ST                        0.009723
BLOCK E 6TH ST                        0.008994
BLOCK BURTON DR                       0.008751
BLOCK GUADALUPE ST                    0.008508
BLOCK E OLTORF ST                     0.008264
BLOCK S 1ST ST                        0.008021
BLOCK W WILLIAM CANNON DR             0.007778
BLOCK N I 35 SB SVC RD                0.007535
BLOCK CROSSING PL                     0.007535
BLOCK E 4TH ST                        0.007292
BLOCK MANCHACA RD                     0.007049
BLOCK NORTHGATE BLVD                  0.006563
BLOCK RUTLAND DR                      0.006320
BLOCK N FM 62

### Question 3. In what location type is violent crime most likely to occur? 
Note: For non-capital murder, and rape, I am only including results that >= 1% of the total.

In [6]:
print('----------------------------------------------------')
print('Location types where violent crimes most often occur')
print('----------------------------------------------------')
print('Murder')
print('----------------------------------')
display(df_mur.location_type.value_counts().head(9))
print('----------------------------------')
display(df_mur.location_type.value_counts(normalize=True).head(9))
print('----------------------------------')
print('Capital Murder')
print('----------------------------------')
display(df_mur_cap.location_type.value_counts())
print('----------------------------------')
display(df_mur_cap.location_type.value_counts(normalize=True))
print('----------------------------------')
print('Rape')
print('----------------------------------')
display(df_rape.location_type.value_counts().head(8))
print('----------------------------------')
display(df_rape.location_type.value_counts(normalize=True).head(8))

----------------------------------------------------
Location types where violent crimes most often occur
----------------------------------------------------
Murder
----------------------------------


RESIDENCE / HOME                163
PARKING LOTS / GARAGE            71
STREETS / HWY / ROAD / ALLEY     71
FIELD / WOODS                    17
OTHER / UNKNOWN                  13
HOTEL / MOTEL / ETC.             10
BAR / NIGHT CLUB                  6
COMMERCIAL / OFFICE BUILDING      6
GAS / SERVICE STATIONS            5
Name: location_type, dtype: int64

----------------------------------


RESIDENCE / HOME                0.425587
PARKING LOTS / GARAGE           0.185379
STREETS / HWY / ROAD / ALLEY    0.185379
FIELD / WOODS                   0.044386
OTHER / UNKNOWN                 0.033943
HOTEL / MOTEL / ETC.            0.026110
BAR / NIGHT CLUB                0.015666
COMMERCIAL / OFFICE BUILDING    0.015666
GAS / SERVICE STATIONS          0.013055
Name: location_type, dtype: float64

----------------------------------
Capital Murder
----------------------------------


RESIDENCE / HOME                        47
PARKING LOTS / GARAGE                   14
STREETS / HWY / ROAD / ALLEY             9
COMMERCIAL / OFFICE BUILDING             3
BAR / NIGHT CLUB                         2
OTHER / UNKNOWN                          1
FIELD / WOODS                            1
CHURCH / SYNAGOGUE / TEMPLE / MOSQUE     1
GROCERY / SUPERMARKET                    1
Name: location_type, dtype: int64

----------------------------------


RESIDENCE / HOME                        0.594937
PARKING LOTS / GARAGE                   0.177215
STREETS / HWY / ROAD / ALLEY            0.113924
COMMERCIAL / OFFICE BUILDING            0.037975
BAR / NIGHT CLUB                        0.025316
OTHER / UNKNOWN                         0.012658
FIELD / WOODS                           0.012658
CHURCH / SYNAGOGUE / TEMPLE / MOSQUE    0.012658
GROCERY / SUPERMARKET                   0.012658
Name: location_type, dtype: float64

----------------------------------
Rape
----------------------------------


RESIDENCE / HOME                      2507
HOTEL / MOTEL / ETC.                   345
STREETS / HWY / ROAD / ALLEY           329
OTHER / UNKNOWN                        297
PARKING LOTS / GARAGE                  202
FIELD / WOODS                           87
BAR / NIGHT CLUB                        73
DRUG STORE / DR. OFFICE / HOSPITAL      67
Name: location_type, dtype: int64

----------------------------------


RESIDENCE / HOME                      0.615366
HOTEL / MOTEL / ETC.                  0.084683
STREETS / HWY / ROAD / ALLEY          0.080756
OTHER / UNKNOWN                       0.072901
PARKING LOTS / GARAGE                 0.049583
FIELD / WOODS                         0.021355
BAR / NIGHT CLUB                      0.017919
DRUG STORE / DR. OFFICE / HOSPITAL    0.016446
Name: location_type, dtype: float64

**Summary of Violent Crimes**

For NON-CAPITAL MURDER, analysis indicates 3 separate addresses which played host to at least 3 counts each, since 2003: 

1.	4700 E Riverside Dr
2.	8610 N Lamar Blvd 
3.	8800 N IH 35, Svrd SB. 

A Google search of the addresses indicated the first address home to the Tempo Apartment Complex. The second is the location of what appears to be a strip mall. The third is home to the Starburst Apartment Complex. 

In addition, 2 separate addresses played host to at least 2 counts of CAPITAL MURDER each: 

1.	815 W Slaughter Lane
2.	7000 Decker Lane 

***Note: Remember that CAPITAL murder means that the defendant(s), based on his/her/their motivations and actions during his/her/their crime commission, makes him/her/them automatically eligible for the death sentence, under Texas law.***

Interestingly, rape occurred, or was reported as having occurred, an astonishing 51 separate times at the 900 Block of E 32nd Street, since 2003.

***Note: Seton Hospital is located here. Why is this? Possibly, because rape often coincides with other forms of violence against the victim(s) and so the victim(s) were reporting from a hospital bed?***  

At any rate, further research is necessary to determine what the reason(s) are with more assurance. 

Further, when we break down violent crimes by the street on which they occurred, we see that most non-capital murders occurred on E Riverside Dr, a total of 13 counts. However, 4 separate capital murders occurred on N Lamar Blvd, as well as 10 separate non-capital murders, which makes N Lamar Blvd the number 1 hotspot for murder in general.   

In terms of individual streets, rape occurred 95 separate times somewhere along N IH 35 SVRD SB, specifically. N Lamar Blvd comes in at number 2, with 80 separate counts of rape at one point or another along that route. 

Another street of note is Wickersham Ln which has hosted 53 separate counts of rape and 4 separate counts of non-capital murder, since 2003. 

Finally, we see that home is not necessarily where one is the safest…

42.56% of non-capital murder, nearly 60% of capital murder, and over 61% of rape all occur within a residence or home. 

As a member of the LGBT Community, I particularly remember an incident from 2013, in which a man, David Villareal, brought home another man he met at Rain Nightclub on 4th Street. The man he brought home, Matthew Bacon, proceeded to bludgeon and eviscerate him to death, before stealing his belongings and making his getaway in David’s vehicle. Matthew has since received 60 years for his crimes, for which he avoided the death sentence by confessing. 

The lesson, ladies and gentlemen: BE CAREFUL WHAT STRANGER(S) YOU INVITE INTO YOUR HOME!!


 


In [8]:
frames = [df_mur, df_mur_cap, df_rape]

df = pd.concat(frames)
df.sort_index(inplace=True)

In [16]:
df.address.value_counts()

900 BLOCK E 32ND ST                     51
700 BLOCK E 8TH ST                      29
1300 BLOCK CROSSING PL                  28
12200 BLOCK N MOPAC EXPY SVRD NB        18
4100 BLOCK GUADALUPE ST                 17
300 BLOCK E 6TH ST                      17
1600 BLOCK WICKERSHAM LN                16
500 BLOCK E 4TH ST                      15
1500 BLOCK WICKERSHAM LN                15
600 BLOCK E 15TH ST                     15
900 BLOCK W BEN WHITE BLVD SVRD EB      14
1500 BLOCK GROVE BLVD                   13
2100 BLOCK BURTON DR                    12
400 BLOCK E WILLIAM CANNON DR           11
9400 BLOCK N IH 35 SVRD SB              10
1100 BLOCK GUNTER ST                    10
9300 BLOCK NORTHGATE BLVD               10
1600 BLOCK ROYAL CREST DR               10
11100 BLOCK RESEARCH BLVD SVRD NB       10
5300 BLOCK N IH 35 SVRD SB              10
4400 BLOCK E OLTORF ST                  10
200 BLOCK LAVACA ST                     10
2700 BLOCK S IH 35 SVRD NB              10
9000 BLOCK 